docs/external-node.md

```diff
@@ -0,0 +1,532 @@
+where `antrea-agent` is running. 
+
+For the detail configuration options when running `antrea-agent`on an external
+Node, please refer to the [sample configuration file](../build/yamls/externalnode/conf/antrea-agent.conf).
+
+# The ExternalNode resource
+
+A typical `ExternalNode` resource example:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: ExternalNode
+metadata:
+  name: vm1
+  namespace: vm-ns
+  labels:
+    role: db
+spec:
+  interfaces:
+    - ips: [ "172.16.100.3" ]
+      name: ""
+```
+
+## Name and Namespace
+
+The `name` field in an `ExternalNode` is unique, which is the key to tell
+`antrea-agent` where it is running on. It is also used as the key in the
+connection from `antrea-agent` to `antrea-controller`. The `ExternalNode` name
+is given to `antrea-agent` via an environment variable `NODE_NAME`, and
+`antrea-agent` will use the hostname to find the `ExternalNode` resource if it
+is not set.
+
+`ExternalNode` resource is working in `namespace` scope. The `namespace` is
+given to `antrea-agent` with option `externalNodeNamespace` in [antrea-agent.conf](../build/yamls/externalnode/conf/antrea-agent.conf).
+
+```yaml
+externalNodeNamespace: vm-ns
+```
+
+## Interfaces
+
+The `interfaces` field specifies the list of the network interfaces expected to
+be guarded by Antrea NetworkPolicy. One interface at least is required. The
+`name` or `ips` can be used to identify the target interface. **The field `ips`
+must be provided in the CRD**, but `name` is optional. Multiple IPs on a single
+interface is supported.
+
+`antrea-controller` creates an `ExternalEntity` for each interface whenever it
+finds an `ExternalNode is created. The created `ExternalEntity` has the
+following characteristics:
+
+- It is configured within the same namespace as the `ExternalNode`.
+- The `name` is generated according to the following principles:
+    - Use the `ExternalNode` name directly, if there is only one interface, and
+      interface name is not specified.
+    - Use the format `$ExternalNode.name-$hash($interface.name)[:5]` for other
+      cases.
+- The `externalNode` field is set with the `ExternalNode` name.
+- The `owner` is referring to the `ExternalNode` resource.
+- All labels added on `ExternalNode` are copied to the `ExternalEntity`.
+- One `Endpoint` is created per IP in the interface, and every Endpoint's name
+  uses the interface name if it is set.
+
+A sample of the `ExternalEntity` created for the above `ExternalNode` interface
+is,
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha2
+kind: ExternalEntity
+metadata:
+  labels:
+    role: db
+  name: vm1
+  namespace: vm-ns
+  ownerReferences:
+    - apiVersion: v1alpha1
+      kind: ExternalNode
+      name: vm1
+      uid: 99b09671-72da-4c64-be93-17185e9781a5
+  resourceVersion: "5513"
+  uid: 5f360f32-7806-4d2d-9f36-80ce7db8de10
+spec:
+  endpoints:
+    - ip: 172.16.100.3
+  externalNode: vm1
+```
+
+Note: **Only one interface is supported for Antrea v1.8**.
```

Comment:
this should be mentioned earlier, when describing the `ExternalNode` resource, and not after introducing `ExternalEntity`